components/site/src/sass/serde.rs

```diff
@@ -0,0 +1,481 @@
+    {
+        self.output += ":";
+        value.serialize(&mut **self)
+    }
+
+    fn end(self) -> SassMapSerializerResult {
+        self.output += ")";
+        Ok(())
+    }
+}
+
+// Same thing but for structs, where the keys are static string field names.
+impl<'a> serde::ser::SerializeStruct for &'a mut SassMapSerializer {
+    type Ok = ();
+    type Error = SassMapSerializerError;
+
+    fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> SassMapSerializerResult
+    where
+        T: ?Sized + Serialize,
+    {
+        if !self.output.ends_with('(') {
+            self.output += ",";
+        }
+        key.serialize(&mut **self)?;
+        self.output += ":";
+        value.serialize(&mut **self)
+    }
+
+    fn end(self) -> SassMapSerializerResult {
+        self.output += ")";
+        Ok(())
+    }
+}
+
+// Same thing but for struct variants, where we ignore the variant name.
+impl<'a> serde::ser::SerializeStructVariant for &'a mut SassMapSerializer {
+    type Ok = ();
+    type Error = SassMapSerializerError;
+
+    fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> SassMapSerializerResult
+    where
+        T: ?Sized + Serialize,
+    {
+        if !self.output.ends_with('(') {
+            self.output += ",";
+        }
+        key.serialize(&mut **self)?;
+        self.output += ":";
+        value.serialize(&mut **self)
+    }
+
+    fn end(self) -> SassMapSerializerResult {
+        self.output += ")";
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_primitives() {
+        verify_serialization(true, "true");
+        verify_serialization(64, "64");
+        verify_serialization(-123, "-123");
+        verify_serialization(567.89, "567.89");
+        verify_serialization('t', "\"t\"");
+        verify_serialization("abc", "\"abc\"");
+        verify_serialization(Option::<i32>::None, "null");
+        verify_serialization((), "null");
+    }
+
+    #[test]
+    fn test_arrays() {
+        verify_serialization(&[123, 456, 789], "[123,456,789]");
+        verify_serialization((123, 456, 789), "[123,456,789]");
+    }
+
+    #[test]
+    fn test_struct() {
+        #[derive(Serialize)]
+        struct Inner {
+            a: Vec<i32>,
+            b: (String, f64),
+        }
+
+        #[derive(Serialize)]
+        struct Outer {
+            a: i32,
+            b: String,
+            c: Inner,
+        }
+
+        let val = Outer {
+            a: 42,
+            b: "abc".to_string(),
+            c: Inner { a: vec![6, 7, 8], b: ("def".to_string(), 123.45) },
+        };
+
+        let expected = "(\"a\":42,\"b\":\"abc\",\"c\":(\"a\":[6,7,8],\"b\":[\"def\",123.45]))";
+        verify_serialization(val, expected);
+    }
+
+    fn verify_serialization<T: Serialize>(val: T, expected: &str) {
+        let mut ser = SassMapSerializer::default();
+        val.serialize(&mut ser).unwrap();
+        assert_eq!(ser.output, expected);
+    }
+}
```

Comment:
cc @connorskees Do you think it's something that would be interesting to have in grass directly? I can see injecting data being useful for more than Zola

Reply:
i think the suggestion was to move the sass map serializer into grass, so that anything that implements `serde::Serialize` can be converted to a sass map literal string.  this would enable the functionality i've built in zola.

i don't have a feel for whether or not that functionality has other use cases, so i'm happy either way (that remaining in zola or being moved to grass).

Reply:
Ah. I would probably not implement this in grass directly to avoid the serde dependency, even if optional. I would say the more common solution would be to just define the config in Sass directly, which is what the major Sass libraries do AFAIK. 

I think zola is a special case here in needing to serialize TOML to Sass. Though I do think this code is quite cool. If there is another project out there that wants this functionality, I would be open to bringing it into grass-proper. It just currently seems a bit too niche.

Reply:
ok I'm going to leave the functionality in Zola for now